generators/chipyard/src/main/scala/example/dsptools/DspBlocks.scala

```diff
@@ -0,0 +1,146 @@
+package chipyard.example
+
+import chisel3._
+import chisel3.util._
+import dspblocks._
+import dsptools.numbers._
+import freechips.rocketchip.amba.axi4stream._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.subsystem._
+
+/**
+  * The memory interface writes entries into the queue.
+  * They stream out the streaming interface
+  * @param depth number of entries in the queue
+  * @param streamParameters parameters for the stream node
+  * @param p
+  */
+abstract class WriteQueue
```

Comment:
Is everything in the file specific to the example? Is it's there is nothing specific to fir/passthrough here, it might be better to move it out of example into chipyard/src/main/scala

Reply:
It mixes in dsptools traits, so although it is not fir/passthrough specific, it is dsptools specific.

Reply:
I agree it's DSPtools specific. I'm just not sure why it's under "example". To me it doesn't seem like an example, but rather part of the framework

Reply:
I think they're part of the example. You shouldn't really use them in a real design, I think you should use a real DMA.

Reply:
So that should probably be clarified in the docs (evidently, I did not understand that)